Source/Core/DolphinQt/Config/Mapping/GCPadWiiUConfigDialog.cpp

```diff
@@ -24,28 +24,31 @@ GCPadWiiUConfigDialog::GCPadWiiUConfigDialog(int port, QWidget* parent)
   ConnectWidgets();
 }
 
+GCPadWiiUConfigDialog::~GCPadWiiUConfigDialog()
+{
+  GCAdapter::SetAdapterCallback(nullptr);
+}
+
 void GCPadWiiUConfigDialog::CreateLayout()
 {
   setWindowTitle(tr("GameCube Adapter for Wii U at Port %1").arg(m_port + 1));
 
-  const bool detected = GCAdapter::IsDetected();
   m_layout = new QVBoxLayout();
-  m_status_label = new QLabel(detected ? tr("Adapter Detected") : tr("No Adapter Detected"));
+  m_status_label = new QLabel();
   m_rumble = new QCheckBox(tr("Enable Rumble"));
   m_simulate_bongos = new QCheckBox(tr("Simulate DK Bongos"));
   m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
 
+  UpdateAdapterStatus();
+
+  GCAdapter::SetAdapterCallback(
+      [this] { QMetaObject::invokeMethod(this, &GCPadWiiUConfigDialog::UpdateAdapterStatus); });
```

Comment:
You don't need `invokeMethod` or an anonymous function here, `&GCPadWiiUConfigDialog::UpdateAdapterStatus` can be passed directly to `SetAdapterCallback` as it has the appropriate params and return type.